.changeset/purple-lemons-yawn.md

```diff
@@ -0,0 +1,8 @@
+---
+"@guild-docs/algolia": minor
+"guild-docs": minor
+---
+
+Support TypeScript ESM modules (`"module": "node16"` and `"moduleResolution": "node16"`).
```

Comment:
This is kind of wrong. This package does not support ESM. It only supports "fake ESM" as webpack and other bundlers do. It will fail when being imported in a context that implements ESM according to the spec.